net/netdev.go

```diff
@@ -0,0 +1,159 @@
+package net
+
+import (
+	"fmt"
+	"net"
+	"os"
+	"regexp"
+	"strconv"
+	"strings"
+
+	"github.com/vishvananda/netlink"
+	"github.com/vishvananda/netns"
+)
+
+var netdevRegExp = regexp.MustCompile(`^([^ ]+?) ([^ ]+?) \[([^]]*)\]$`)
+
+type Dev struct {
+	Name  string
+	MAC   net.HardwareAddr
+	CIDRs []*net.IPNet
+}
+
+func (d Dev) String() string {
+	return fmt.Sprintf("%s %s %s", d.Name, d.MAC, d.CIDRs)
+}
+
+func ParseNetDev(netdev string) (Dev, error) {
```

Comment:
As mentioned elsewhere, this (and the associated regexp/`Dev.String()`) can go away if `list-netdevs` prints JSON